WooCommerce/Classes/POS/ViewHelpers/TotalsViewHelper.swift

```diff
@@ -73,4 +73,12 @@ final class TotalsViewHelper {
             return true
         }
     }
+
+    func shouldShowTotalDiscountField(cart: Cart, orderTotals: PointOfSaleOrderTotals?) -> Bool {
+        guard cart.coupons.isNotEmpty else {
+            return false
+        }
+
+        return orderTotals == nil || orderTotals?.discountTotal != nil
+    }
```

Comment:
Should we add comments here regarding the logic? Or perhaps declare them explicitly? Eg:
```
let hasCoupons = cart.coupons.isNotEmpty
let hasDiscounts = orderTotals?.discountTotal != nil
... 
```

Question: The comment on e9f4ea56c58d307fb6d546c28b6fd4b239712fd2 shares that the discounts have to be non-zero, but this checks for non-nil discounts. Does this cover the same case? Or should be handled separately?

Reply:
> Question: The comment on https://github.com/woocommerce/woocommerce-ios/commit/e9f4ea56c58d307fb6d546c28b6fd4b239712fd2 shares that the discounts have to be non-zero, but this checks for non-nil discounts. Does this cover the same case? Or should be handled separately?

Yes, in this case, it's nil when the discount is zero. But it's not really clear. I'll try to organize and name it in a different.